nupic/embodied/envs/wrappers.py

```diff
@@ -0,0 +1,539 @@
+        self._actions = acts
+        self.action_space = gym.spaces.Discrete(len(self._actions))
+
+    def action(self, a):
+        mask = np.zeros(self._num_buttons)
+        for i in self._actions[a]:
+            mask[i] = 1
+        return mask
+
+
+class FrameSkip(gym.Wrapper):
+    def __init__(self, env, n):
+        gym.Wrapper.__init__(self, env)
+        self.n = n
+
+    def step(self, action):
+        done = False
+        totrew = 0
+        for _ in range(self.n):
+            ob, rew, done, info = self.env.step(action)
+            totrew += rew
+            if done:
+                break
+        return ob, totrew, done, info
+
+
+def make_mario_env(crop=True, frame_stack=True, clip_rewards=False):
+    assert clip_rewards is False
+    import retro
+    from baselines.common.atari_wrappers import FrameStack
+
+    # gym.undo_logger_setup()
+    env = retro.make("SuperMarioBros-Nes", "Level1-1")
+    buttons = env.buttons
+    env = MarioXReward(env)
+    env = FrameSkip(env, 4)
+    env = ProcessFrame84(env, crop=crop)
+    if frame_stack:
+        env = FrameStack(env, 4)
+    env = LimitedDiscreteActions(env, buttons)
+    return env
+
+
+class OneChannel(gym.ObservationWrapper):
+    def __init__(self, env, crop=True):
+        self.crop = crop
+        super(OneChannel, self).__init__(env)
+        assert env.observation_space.dtype == np.uint8
+        self.observation_space = gym.spaces.Box(
+            low=0, high=255, shape=(84, 84, 1), dtype=np.uint8
+        )
+
+    def observation(self, obs):
+        return obs[:, :, 2:3]
+
+
+class RetroALEActions(gym.ActionWrapper):
+    def __init__(self, env, all_buttons, n_players=1):
+        gym.ActionWrapper.__init__(self, env)
+        self.n_players = n_players
+        self._num_buttons = len(all_buttons)
+        bs = [-1, 0, 4, 5, 6, 7]
+
+        def update_actions(old_actions, offset=0):
+            actions = []
+            for b in old_actions:
+                for button in bs:
+                    action = []
+                    action.extend(b)
+                    if button != -1:
+                        action.append(button + offset)
+                    actions.append(action)
+            return actions
+
+        current_actions = [[]]
+        for i in range(self.n_players):
+            current_actions = update_actions(current_actions, i * self._num_buttons)
+        self._actions = current_actions
+        self.action_space = gym.spaces.Discrete(len(self._actions))
+
+    def action(self, a):
+        mask = np.zeros(self._num_buttons * self.n_players)
+        for i in self._actions[a]:
+            mask[i] = 1
+        return mask
+
+
+class NoReward(gym.Wrapper):
+    def __init__(self, env):
+        gym.Wrapper.__init__(self, env)
+
+    def step(self, action):
+        ob, rew, done, info = self.env.step(action)
+        return ob, 0.0, done, info
+
+
+def make_multi_pong(frame_stack=True):
```

Comment:
We should move all functions to after the classes (or before). I believe the autoformatter will do that for you.